torch/distributed/rendezvous.py

```diff
@@ -147,6 +150,46 @@ def _env_error(var):
     raise RuntimeError("Unable to perform rerendezvous using env:// method")
 
 
+def _mpi_rendezvous_handler(url):
+    def _error(msg):
+        return _rendezvous_error("mpi:// rendezvous: " + msg)
+
+    if not url.startswith("mpi://"):
+        raise _error("URL must start with `mpi://`")
+
+    if not is_mpi_available():
+        raise _error("MPI backend not available")
+
+    result = urlparse(url)
+    query = dict(pair.split("=") for pair in filter(None, result.query.split("&")))
+    port = int(query.get("port", 0))
+
+    def get_local_address(port=0):
```

Comment:
Heh can't wait to see sysadmins of managed scientific computing environments to wonder why are there so many jobs pinging 8.8.8.8. Is there really no other way?